### ISSUE TITLE
valid_docs rule failing on closure return type

### ISSUE THREAD

**Issue by aamctustwo** [realm/SwiftLint#282]
The `valid_docs` rule throws a false positive when the return type is a closure. Currently tested on SwiftLint 0.5.2

**Example:**
```swift
struct Foo {
    /**
     Bar description
     - returns: Closure return.
     */
    func bar() -> (Int -> String) {
        return { (baz: Int) -> String in
            return "\(baz)"
        }
    }
}
```

**Comment by jschmid** [comment]
Same for tuples:

```swift
struct Foo {
    /**
     Bar description
     - returns: A pair.
     */
    func bar() -> (Int, String) {
        return (1, "Baz")
    }
}
```

**Comment by jpsim** [comment]
Thanks for filing this. This is something we should certainly fix. Help is appreciated.

**Comment by jpsim** [comment]
This is fixed as of SwiftLint 0.7.1. Thanks for reporting!
